### ISSUE TITLE
Missing SIGCHLD from WebKitPluginProcess

### ISSUE THREAD

**Issue by Ygrex** [aidanholm/luakit#85]
What I do:
1. break plugin scanning:
   1.1. "rm -f ~/.cache/webkitgtk/plugins"
   1.2. "chmod 0000 /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so"
2. start luakit: "luakit -U"

What I see:

error message from plugin scanning, warning from glib, suspicious exit code:

```
Error loading module '/usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so': /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so: cannot open shared object file: Permission denied

(luakit:348): GLib-WARNING **: In call to g_spawn_sync(), exit status of a child process was requested but ECHILD was received by waitpid(). Most likely the process is ignoring SIGCHLD, or some other thread is invoking waitpid() with a nonpositive first argument; either behavior can break applications that use g_spawn_sync either directly or indirectly.
Error scanning plugin /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so, /usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitPluginProcess returned 32695 exit status
```

What I expect:

error message from plugin scanning, no glib warnings, correct exit code shown

could not find a stable way to reproduce, but here is glib warning:

```
$ ./luakit -U
Stylesheet directory '/home/ygrex/.local/share/luakit/styles/' doesn't exist, not loading user styles...
luakit web process: connecting to /home/ygrex/.cache/luakit/socket
luakit web process: connected
luakit web process: Lua initializing...
luakit web process: Lua initialized
luakit web process: ready for messages
Error loading module '/usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so': /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so: cannot open shared object file: Permission denied

(luakit:348): GLib-WARNING **: In call to g_spawn_sync(), exit status of a child process was requested but ECHILD was received by waitpid(). Most likely the process is ignoring SIGCHLD, or some other thread is invoking waitpid() with a nonpositive first argument; either behavior can break applications that use g_spawn_sync either directly or indirectly.
Error scanning plugin /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so, /usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitPluginProcess returned 32695 exit status
```

here is the output that is shown most of the time, no glib warning but exit code is 256:

```
$ ./luakit -U
Stylesheet directory '/home/ygrex/.local/share/luakit/styles/' doesn't exist, not loading user styles...
luakit web process: connecting to /home/ygrex/.cache/luakit/socket
luakit web process: connected
luakit web process: Lua initializing...
luakit web process: Lua initialized
luakit web process: ready for messages
Error loading module '/usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so': /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so: cannot open shared object file: Permission denied
Error scanning plugin /usr/lib/mozilla/plugins/libgnome-shell-browser-plugin.so, /usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/WebKitPluginProcess returned 256 exit status
```

**Comment by aidanholm** [comment]
This looks like a bug in WebKit GTK. As far as I can tell WebKit plugins aren't accessed or modified from luakit at all.

**Comment by Ygrex** [comment]
sigchld() invokes waitpid(-1), the warning never shows up if this handler disabled

**Comment by aidanholm** [comment]
Ah. The original purpose of this signal handler (i.e. why there might be child processes in the first place) is unclear. I'll do some testing and remove it if there don't appear to be any zombies left over.

**Comment by aidanholm** [comment]
Seems fine; fix pushed at db42a6e; thanks!
